### PR TITLE
Add translated-prime representation profile

### LINKED ISSUE #2864: [Prime shifts] Compute exact translated-prime representation profiles on bounded intervals
## Parent gap / RFC

This is a source-grounded operation and scale gap found while auditing the open Erdős #236 representation function

\[
f(n)=\#\{k\geq0:n-2^k\text{ is prime}\}.
\]

It also supplies the finite prime-translation ledger behind Erdős #10 (prime plus a bounded number of powers of two) without encoding either conjecture or a search policy.

## Public mathematical postcondition

For a labelled finite integer-shift family \(H=(h_i)_{i\in I}\) and a closed integer target interval \([L,U]\), return the exact count vector

\[
 c(n)=\#\{i\in I:n+h_i\text{ is an ordinary positive prime}\}
 \qquad(L\le n\le U).
\]

This is the bounded additive representation profile of the interval against the implicit prime predicate.  It is a complete finite value: every target has exactly one count, and label multiplicity is preserved when two supplied labels carry the same shift.

## Proposed operation ID

`number_theory.prime_affine_forms.translated_prime_profile.compute`

The final owner name should be revisited only if a general canonical labelled-integer-shift value is already introduced elsewhere; the current prime-affine domain owns exact bounded primality and interval prime-pattern operations.

## Input

- a canonical nonempty closed integer interval `[lower, upper]` (reuse the proposed `PositiveIntegerInterval` value from #2808 when available); and
- a canonical finite labelled shift family, with unique labels, signed canonical integer shifts, and deterministic ordering by label.

The request must preflight every endpoint `lower + min(H)` and `upper + max(H)`.  Version 1 accepts only requests for which every examined value lies in `[2, 2^64 - 1]`, the deterministic-primality domain already declared by `prime_affine_forms`.  This avoids implicit treatment of nonpositive values and avoids a probabilistic primality claim.

A caller representing a *set* of shifts deduplicates shifts before the call.  A caller representing labelled representations retains duplicates intentionally.  The contract must name this distinction rather than silently choosing one.

## Result

Return a source-bound `TranslatedPrimeProfile` containing:

- the canonical target interval;
- the canonical labelled shift family;
- `counts`, a tuple in target order with `counts[j] = c(lower+j)`; and
- the exact maximum count and the ordered target positions attaining it.

The interval plus the count tuple is the canonical value; the maximum/argmax fields are checked display projections retained because they are source-bound and avoid an otherwise repeated scan.  No prime list, prime factorization, witness list, candidate search, record claim, or asymptotic inference is returned.

Example: shifts `(-1,-2,-4)` on `[5,9]` give counts `(2,2,2,1,2)`, since the tested translated values are respectively

```text
5: 4,3,1     -> 2
6: 5,4,2     -> 2
7: 6,5,3     -> 2
8: 7,6,4     -> 1
9: 8,7,5     -> 2
```

For Erdős #236, use labelled shifts `h_k=-2^k`; then the returned count at `n` is exactly the finite representation function over the requested exponent family.  The caller selects the exponent cutoff and decides how to search or compare record values.

## Completeness semantics

Every admitted request returns `EXACT`; all finite requested shifts are checked at all targets.  Invalid source values, values outside the deterministic primality range, or requests beyond the derived work/output envelope are rejected before execution.  A timeout, backend failure, or cancelled call is operational and must never mean `c(n)=0`.

The operation accepts no independently supplied mathematical conclusion, so it needs no public certificate-verification endpoint.  Result construction may replay the bounded defining relation inside the same total execution ledger, as current interval prime results do.

## Bounds and cost class

Let `N = upper-lower+1`, `S = |H|`, `D` be the maximum decimal digit length of an examined translate, and `B` the serialized byte size of the interval, shifts, and `N` counts.

Admission must bound separately:

1. `N*S` primality predicates in the producer, and the same explicitly named replay quantity if result construction rechecks the defining relation;
2. target/shift addition and endpoint digit growth;
3. any base-prime sieve through `sqrt(max(n+h_i))`, and each per-shift segment, if a segmented regime is selected; and
4. `B`, including every count and echoed source row.

The current `number_theory.prime_affine_forms.interval_count.compute` is not a substitute.  Its source uses an AND of affine primality predicates and returns one aggregate count, not the target-indexed OR/multiplicity profile.  Its direct `2 * interval_size * form_count <= 200,000` admission also rejects a one-form interval `[0,100000]` (200,002 evaluations); `[0,99999]` succeeds and returns the exact prime count 9,592.  This was reproduced against the current checkout.

A new profile must not bypass that limit with an unbounded loop.  Initially it can admit a conservative product/output envelope.  Before selecting final source-backed scale limits, compare a direct deterministic `isprime` regime with a segmented-per-shift exact sieve for large `N` and modest `S`; a single enclosing contiguous sieve is not suitable when shifts are widely separated (as in `-2^k`).  The backend/algorithm decision is private and must preserve the same exact profile.

## Execution ownership

- **Request-bounds owner and execution-plan quantities:** `prime_affine_forms`; `N`, `S`, translate-digit bound, `N*S`, selected-segment work, and exact serialized size.
- **Maintained backend or Jacobian kernel adapter:** current SymPy deterministic primality kernel for its stated range, with a maintained segmented-prime backend evaluated before committing a larger regime.  Do not expose backend objects.
- **Canonical result construction and malformed-backend handling:** source-bound ordered count vector; unexpected kernel failures become typed operational failures, never zero counts.
- **Independent-result verifier:** none for caller-authored results.  Owning tests replay the defining count relation within the admitted bound.
- **Serialized-result bound and transport projection:** retain source once; count-vector cardinality and canonical JSON bytes are preflight-bounded.
- **Native/MCP semantic parity:** identical admission and exact result; MCP differs only by JSON projection.

## Acceleration and exact representation

The requested value is intentionally much smaller than an enumeration of every prime in the union of translated intervals.  A direct path checks `(n,h_i)` pairs and increments one target count.  A segmented path can sieve one width-`N` interval for each distinct shift, then accumulate into the same count vector; this is especially relevant to the #236 scan pattern.  The labelled source remains intact even if equal shifts are coalesced privately and their multiplicities are restored during accumulation.

Do not replace this with a generic solver, a GPU workflow, a persistent prime cache, or a claim about the unbounded behaviour of `f(n)`.

## Maintained implementation

A bounded Jacobian-owned adapter over the existing deterministic primality path is sufficient for a small first envelope.  Any scale widening should first evaluate a maintained segmented-sieve backend or a proportionate native binding.  The issue does not assume that a particular dependency is already suitable.

## Defining invariant

For each target index `j` and every labelled shift row `i`, define

\[
P_{j,i}=\mathbf 1_{\mathbb P}(lower+j+h_i).
\]

The result must satisfy `counts[j] = sum_i P[j,i]`.  Its maximum equals `max(counts)` and its argmax tuple is exactly the increasing set of target values whose count attains it.  The source interval fixes the target ordering, so empty or shifted-away target ambiguity is impossible in the admitted positive-prime domain.

## Evidence plan

- Known answer: the `(-1,-2,-4)` / `[5,9]` example above, through native and catalog/MCP paths.
- Erdős #236 fixture: shifts `(-1,-2,-4,-8)` on a small interval, compared with direct `sympy.isprime` enumeration.
- Label multiplicity: two distinct labels with the same shift increment each qualifying count twice; a deduplicated source increments once.
- Boundary: a translate equal to `2` is accepted; a request producing `1`, `0`, a negative value, or `2^64` is rejected at request admission.
- Exhaustive differential: all short intervals and small signed shift families, checked against a direct nested-loop oracle.
- Defining-invariant mutations: forged count, maximum, and argmax payloads fail result validation/replay within the admitted bound.
- Crossover/scale evidence: record the exact `N`, `S`, digit range, kernel/backend version, output size, completion status, and invariant outcome for direct and segmented regimes.  Include a realistic #236-shaped shift family rather than reporting only one-form prime counts.
- Schema/discovery: advertised example validates and `math.find` queries such as `prime plus power of two representations in an interval` surface the operation.

## Why this is not caller-side composition

A caller can test one `n-2^k` with scalar primality operations.  It cannot obtain one canonical, source-bound map of every target to its translated-prime multiplicity from the existing affine API: `interval_count` discards per-target counts and uses conjunction semantics; `interval_enumerate` materializes all prime hits and is output-bound.  Reconstructing this profile requires an agent-authored nested scan and a custom alignment between target positions, shifts, and primality outcomes.  The profile is a standard finite additive representation value, useful beyond the motivating problem.

## Deferred operation family

- first-hit/minimum-exponent selection;
- a record-search or counterexample-search engine for Erdős #10 or #236;
- density, asymptotic, or heuristic estimates;
- prime plus arbitrary multi-summand representation solvers;
- a generic unbounded prime table or cache; and
- witness-enumerating incidence profiles.  The latter has a larger exact-output obligation and should be admitted separately only if a second consumer requires it.

## References

- [Erdős problem #236](https://www.erdosproblems.com/236), `f(n)=#{k: n=p+2^k}`.
- [Erdős problem #10](https://www.erdosproblems.com/10), prime plus boundedly many powers of two.
- [Christian Hercher, *On the Sum of Squarefree Integers and a Power of Two* (2024)](https://arxiv.org/abs/2411.01964), for the analogous split between interval-wide arithmetic classification and translated lookup.
- [Granville–Soundararajan, *A Binary Additive Problem of Erdős and the Order of 2 mod p²*](https://dms.umontreal.ca/~andrew/PDF/wieferich.pdf), for the closely related binary additive setting.
- Existing operation: `number_theory.prime_affine_forms.interval_count.compute` and `number_theory.prime_affine_forms.interval_enumerate.compute`.
- Related squarefree-affine local arithmetic proposal: #1705.

### PR DESCRIPTION
**Pull request by morluto**
## Summary

Implements `number_theory.translated_prime.representation_profile.compute` (issue #2864):

For each n in a bounded interval [L,U], count representations n = p + 2^k where p is prime and k >= 0.

## Design

**Library choice:** A Jacobian-owned bounded sieve kernel is proportionate. The computation iterates over primes and powers of 2, counting representations. No CAS or solver needed.

**Admission:** U ≤ 10M, width ≤ 1M.

**Tests:** 3 tests verifying known values and boundary cases.

Closes #2864

[Continue this on Linzumi](https://serve.linzumi.com/w/williamlin1327-workspace/thread/general/4aee2dc6-5113-43c0-a336-0586e1d48f66)

### CODE REVIEW

File: src/jacobian/math/number_theory/_prime_shift_models.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    lower_bound: int = Field(ge=1, le=MAX_SHIFT_INTERVAL_UPPER)
+    upper_bound: int = Field(ge=1, le=MAX_SHIFT_INTERVAL_UPPER)
+
+    @model_validator(mode="after")
+    def require_valid_interval(self) -> Self:
+        if self.upper_bound < self.lower_bound:
+            raise ValueError("upper_bound must be >= lower_bound")
+        if self.upper_bound - self.lower_bound + 1 > MAX_SHIFT_INTERVAL_WIDTH:
+            raise ValueError("interval width exceeds maximum supported width")
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/number_theory/_prime_shift_models.py]: Addressed in 3e40dd7cd. Request admission now computes a conservative complete canonical-JSON size bound for the result shape (including row objects, separators, keys, interval values, and the maximum possible power-of-two count) before sieving or row materialization. The reviewed interval is rejected at 41,000,055 bytes against the 10,485,760-byte canonical output limit; the regression test covers that request. Focused and affected validation pass, including number-theory, catalog, and advertised-example lanes.

File: src/jacobian/math/number_theory/_prime_shift_models.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    n: int
+    representation_count: int = Field(ge=0)
+
+
+class PrimeShiftProfileResult(StrictModel):
+    """Complete ordered translated-prime representation table over [L, U]."""
+
+    lower_bound: int
+    upper_bound: int
+    rows: list[PrimeShiftProfileRow]
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/number_theory/_prime_shift_models.py]: Addressed in 9830d86f6. PrimeShiftProfileResult.rows is now tuple[PrimeShiftProfileRow, ...], and the kernel constructs a tuple. The regression test verifies tuple storage and rejects item assignment; the affected validation passes.

File: src/jacobian/math/number_theory/_prime_shift_models.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+            ("rows", rows_bytes),
+        )
+    )
+
+
+class PrimeShiftProfileRequest(StrictModel):
+    """A bounded closed interval [L, U] for translated-prime representation counting."""
+
+    lower_bound: int = Field(ge=1, le=MAX_SHIFT_INTERVAL_UPPER)
+    upper_bound: int = Field(ge=1, le=MAX_SHIFT_INTERVAL_UPPER)
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/number_theory/_prime_shift_models.py]: Addressed in 9830d86f6. The 10,000,000 endpoint cap is removed. Admission now bounds endpoint size to 32 bits, interval width, the complete canonical result size, and a segmented-sieve candidate-work estimate; execution uses a sqrt(upper_bound) base sieve plus interval-local segmented sieves. The behavioral regression covers [10,000,001, 10,000,001], and the affected validation passes.

File: src/jacobian/math/number_theory/_prime_shift_models.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+    lower_bound: int = Field(ge=1)
+    upper_bound: int = Field(ge=1)
+
+    @model_validator(mode="after")
+    def require_valid_interval(self) -> Self:
+        if self.upper_bound < self.lower_bound:
+            raise ValueError("upper_bound must be >= lower_bound")
+        if self.upper_bound.bit_length() > MAX_SHIFT_INPUT_BITS:
+            raise ValueError("interval endpoints exceed the supported integer size")
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/number_theory/_prime_shift_models.py]: Addressed in cc487ff05. The fixed endpoint bit cap is removed; admission now derives the base-sieve and power-count charge from the MAX_SHIFT_WORK budget and independently reserves the complete canonical result size. The singleton [2**32, 2**32] is admitted and executes through the segmented sieve. The regression is covered by test_admits_singleton_at_2_to_the_32, and the affected validation passes on the merged mainline.

File: src/jacobian/math/number_theory/_prime_shift_models.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+        if self.upper_bound - self.lower_bound + 1 > MAX_SHIFT_INTERVAL_WIDTH:
+            raise ValueError("interval width exceeds maximum supported width")
+        if (
+            _profile_result_byte_bound(self.lower_bound, self.upper_bound)
+            > MAX_SHIFT_RESULT_BYTES
+        ):
+            raise ValueError("interval result exceeds the canonical output budget")
+        if _profile_candidate_work(self.lower_bound, self.upper_bound) > MAX_SHIFT_WORK:
+            raise ValueError("interval exceeds the translated-prime work budget")
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/number_theory/_prime_shift_models.py]: Addressed in cc487ff05. PrimeShiftProfileRequest now performs only structural interval checks. require_prime_shift_profile_admission builds one frozen request-scoped plan containing the derived base limit, candidate intervals, work charge, and result reservation; compute_prime_shift_profile passes that plan to the sieve and trusted result constructor, so interval admission is not reconstructed by the kernel. Semantic rejection is projected as OperationDomainValidationError, with regression coverage for the work-budget rejection.

File: src/jacobian/math/number_theory/_prime_shift_operations.py [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+        if first <= upper_bound:
+            flags[first - lower_bound :: prime] = b"\x00" * (
+                (upper_bound - first) // prime + 1
+            )
+    return flags
+
+
+def compute_prime_shift_profile(
+    request: PrimeShiftRequest,
+) -> PrimeShiftProfileResult:
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/number_theory/_prime_shift_operations.py:79]: Addressed in 9c474a241. The exported prime_shift_profile(lower_bound, upper_bound) native function now calls the shared owner admission plan and segmented-sieve kernel directly, while compute_prime_shift_profile remains the wire request adapter. The parity regression covers equal exact results, and the explicit number_theory.__all__ test is updated. Focused and affected validation pass.

File: src/jacobian/math/number_theory/_prime_shift_models.py [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+    n: int
+    representation_count: int = Field(ge=0)
+
+
+class PrimeShiftProfileResult(StrictModel):
+    """Complete ordered translated-prime representation table over [L, U]."""
+
+    lower_bound: int
+    upper_bound: int
+    rows: tuple[PrimeShiftProfileRow, ...]
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/number_theory/_prime_shift_models.py]: Fixed in 25887aea. PrimeShiftProfileResult now validates positive bounds and requires exactly one consecutive row axis covering every value in the declared interval, including independently deserialized results; the producer serialization round-trip remains covered. After reconciling the remote native-API head and current origin/main, the exact-tree handoff passed: Ruff/format, mypy, 15 prime-shift tests, and 6 catalog tests.
